[06/May/2020:12:23:39,222] INFO  BtRuntimeBuilder: Loading standard extension module bt.peer.lan.LocalServiceDiscoveryModule
[06/May/2020:12:23:39,223] INFO  BtRuntimeBuilder: Loading standard extension module bt.peerexchange.PeerExchangeModule
[06/May/2020:12:23:39,224] INFO  BtRuntimeBuilder: Overriding auto-loaded module bt.dht.DHTModule
[06/May/2020:12:23:39,225] INFO  BtRuntimeBuilder: Auto-loading module bt.tracker.http.HttpTrackerModule with default configuration
[06/May/2020:12:23:40,252] INFO  SocketChannelConnectionAcceptor: Opening server channel for incoming connections @ /192.168.0.174:6891
[06/May/2020:12:23:45,309] WARN  MultiTracker: Unexpected error during interaction with the tracker: HttpTracker{baseUri=http://bt3.t-ru.org/ann}
java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:139)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:155)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:284)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:140)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:261)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:165)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:167)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:272)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:124)
	at org.apache.http.impl.execchain.MinimalClientExec.execute(MinimalClientExec.java:213)
	at org.apache.http.impl.client.MinimalHttpClient.doExecute(MinimalHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
	at bt.tracker.http.HttpTracker.sendEvent(HttpTracker.java:150)
	at bt.tracker.http.HttpTracker.access$000(HttpTracker.java:51)
	at bt.tracker.http.HttpTracker$1.query(HttpTracker.java:122)
	at bt.tracker.MultiTracker$1.lambda$query$3(MultiTracker.java:90)
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:125)
	at bt.tracker.MultiTracker$1.query(MultiTracker.java:90)
	at bt.peer.TrackerPeerSource.collectPeers(TrackerPeerSource.java:54)
	at bt.peer.ScheduledPeerSource.lambda$schedulePeerCollection$0(ScheduledPeerSource.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:23:47,578] WARN  MultiTracker: Unexpected error during interaction with the tracker: HttpTracker{baseUri=http://retracker.local/announce}
java.net.UnknownHostException: retracker.local
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at org.apache.http.impl.conn.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:45)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:111)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
	at org.apache.http.impl.execchain.MinimalClientExec.execute(MinimalClientExec.java:183)
	at org.apache.http.impl.client.MinimalHttpClient.doExecute(MinimalHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
	at bt.tracker.http.HttpTracker.sendEvent(HttpTracker.java:150)
	at bt.tracker.http.HttpTracker.access$000(HttpTracker.java:51)
	at bt.tracker.http.HttpTracker$1.query(HttpTracker.java:122)
	at bt.tracker.MultiTracker$1.lambda$query$3(MultiTracker.java:90)
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:125)
	at bt.tracker.MultiTracker$1.query(MultiTracker.java:90)
	at bt.peer.TrackerPeerSource.collectPeers(TrackerPeerSource.java:54)
	at bt.peer.ScheduledPeerSource.lambda$schedulePeerCollection$0(ScheduledPeerSource.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:23:50,251] WARN  ScheduledPeerSource: Peer collection finished with exception in peer source: TrackerPeerSource {MultiTracker{trackerTiers=[[HttpTracker{baseUri=http://bt3.t-ru.org/ann}], [HttpTracker{baseUri=http://retracker.local/announce}]]}}
java.util.concurrent.ExecutionException: bt.BtException: All trackers failed; responses (in chrono order): [bt.tracker.TrackerResponse@2dbadee3, bt.tracker.TrackerResponse@715c6264]
	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at bt.peer.ScheduledPeerSource.schedulePeerCollection(ScheduledPeerSource.java:72)
	at bt.peer.ScheduledPeerSource.update(ScheduledPeerSource.java:60)
	at bt.peer.PeerRegistry.queryPeerSource(PeerRegistry.java:189)
	at bt.peer.PeerRegistry.queryTracker(PeerRegistry.java:167)
	at bt.peer.PeerRegistry.lambda$queryTrackers$4(PeerRegistry.java:148)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.peer.PeerRegistry.queryTrackers(PeerRegistry.java:146)
	at bt.peer.PeerRegistry.lambda$collectAndVisitPeers$3(PeerRegistry.java:116)
	at java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(ConcurrentHashMap.java:4649)
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1082)
	at bt.peer.PeerRegistry.collectAndVisitPeers(PeerRegistry.java:106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: bt.BtException: All trackers failed; responses (in chrono order): [bt.tracker.TrackerResponse@2dbadee3, bt.tracker.TrackerResponse@715c6264]
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:145)
	at bt.tracker.MultiTracker$1.query(MultiTracker.java:90)
	at bt.peer.TrackerPeerSource.collectPeers(TrackerPeerSource.java:54)
	at bt.peer.ScheduledPeerSource.lambda$schedulePeerCollection$0(ScheduledPeerSource.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
[06/May/2020:12:23:50,251] WARN  ScheduledPeerSource: Peer collection finished with exception in peer source: bt.dht.DHTPeerSource@2cee03ec
java.util.concurrent.ExecutionException: bt.BtException: Unexpected error in peer lookup: null. Diagnostics:
==========================
DHT Diagnostics. Type IPV4_DHT
# of active servers / all servers: 1/1
-----------------------
Stats
Reachable node estimate: 2 (0.5)
DB Keys: 0
DB Items: 0
TX sum: 3 RX sum: 2
avg task time/avg 1st result time (ms): 10000/10000
Uptime: PT5.194Ss
RPC stats
### local RPCs
            Method                 REQ |                 RSP               Error             Timeout 

              PING                   3 |                   2                   0                   0 
         FIND_NODE                   0 |                   0                   0                   0 
         GET_PEERS                   0 |                   0                   0                   0 
     ANNOUNCE_PEER                   0 |                   0                   0                   0 
               GET                   0 |                   0                   0                   0 
               PUT                   0 |                   0                   0                   0 
 SAMPLE_INFOHASHES                   0 |                   0                   0                   0 
           UNKNOWN                   0 |                   0                   0                   0 

### remote RPCs
            Method                 REQ |                 RSP              Errors 

              PING                   0 |                   0                   0 
         FIND_NODE                   0 |                   0                   0 
         GET_PEERS                   0 |                   0                   0 
     ANNOUNCE_PEER                   0 |                   0                   0 
               GET                   0 |                   0                   0 
               PUT                   0 |                   0                   0 
 SAMPLE_INFOHASHES                   0 |                   0                   0 
           UNKNOWN                   0 |                   0                   0 
-----------------------
Routing table
buckets: 1 / entries: 2
all   num:2 rep:0 [Home]

-----------------------
RPC Servers
9D0499BD 5BA357B7 A140940E 8963221E 0E15CFF4	bind: /192.168.0.174 consensus: null
rx: 2 tx: 3 active: 1 baseRTT: 8025 loss: 0.490050  loss (verified): 0.500000 uptime: PT5.005S
RTT stats (2samples)  mean:2562.5002 median:125.0 mode:125.0 10tile:125.0 90tile:8025.0
     0 |     50 |    100 |    150 |    200 |    250 |    300 |    350 |    400 |    450 |    500 |    550 |    600 |    650 |    700 |    750 |    800 |    850 |    900 |    950 |   1000 |   1050 |   1100 |   1150 |   1200 |   1250 |   1300 |   1350 |   1400 |   1450 |   1500 |   1550 |   1600 |   1650 |   1700 |   1750 |   1800 |   1850 |   1900 |   1950 |   2000 |   2050 |   2100 |   2150 |   2200 |   2250 |   2300 |   2350 |   2400 |   2450 |   2500 |   2550 |   2600 |   2650 |   2700 |   2750 |   2800 |   2850 |   2900 |   2950 |   3000 |   3050 |   3100 |   3150 |   3200 |   3250 |   3300 |   3350 |   3400 |   3450 |   3500 |   3550 |   3600 |   3650 |   3700 |   3750 |   3800 |   3850 |   3900 |   3950 |   4000 |   4050 |   4100 |   4150 |   4200 |   4250 |   4300 |   4350 |   4400 |   4450 |   4500 |   4550 |   4600 |   4650 |   4700 |   4750 |   4800 |   4850 |   4900 |   4950 |   5000 |   5050 |   5100 |   5150 |   5200 |   5250 |   5300 |   5350 |   5400 |   5450 |   5500 |   5550 |   5600 |   5650 |   5700 |   5750 |   5800 |   5850 |   5900 |   5950 |   6000 |   6050 |   6100 |   6150 |   6200 |   6250 |   6300 |   6350 |   6400 |   6450 |   6500 |   6550 |   6600 |   6650 |   6700 |   6750 |   6800 |   6850 |   6900 |   6950 |   7000 |   7050 |   7100 |   7150 |   7200 |   7250 |   7300 |   7350 |   7400 |   7450 |   7500 |   7550 |   7600 |   7650 |   7700 |   7750 |   7800 |   7850 |   7900 |   7950 |   8000 |   8050 |   8100 |   8150 |   8200 |   8250 |   8300 |   8350 |   8400 |   8450 |   8500 |   8550 |   8600 |   8650 |   8700 |   8750 |   8800 |   8850 |   8900 |   8950 |   9000 |   9050 |   9100 |   9150 |   9200 |   9250 |   9300 |   9350 |   9400 |   9450 |   9500 |   9550 |   9600 |   9650 |   9700 |   9750 |   9800 |   9850 |   9900 |   9950 | 
    0% |     0% |    50% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% | 
-----------------------
Blacklist
{}
-----------------------
Lookup Cache
anchors (0):
buckets (1) / entries (0):

all entries: 0
-----------------------
Tasks
next id: 2
#### active: 
#### queued: 
NodeLookup 2 target:9D0499BD 5BA357B7 A140940E 8963221E 0E15CFF4 todo:0 srv: 9D0499BD 5BA357B7 A140940E 8963221E 0E15CFF4 QUEUED name:Bootstrap: lookup for self




	at java.util.concurrent.FutureTask.report(FutureTask.java:122)
	at java.util.concurrent.FutureTask.get(FutureTask.java:192)
	at bt.peer.ScheduledPeerSource.schedulePeerCollection(ScheduledPeerSource.java:72)
	at bt.peer.ScheduledPeerSource.update(ScheduledPeerSource.java:60)
	at bt.peer.PeerRegistry.queryPeerSource(PeerRegistry.java:189)
	at bt.peer.PeerRegistry.lambda$null$2(PeerRegistry.java:139)
	at java.lang.Iterable.forEach(Iterable.java:75)
	at bt.peer.PeerRegistry.lambda$collectAndVisitPeers$3(PeerRegistry.java:138)
	at java.util.concurrent.ConcurrentHashMap$KeySetView.forEach(ConcurrentHashMap.java:4649)
	at java.util.Collections$UnmodifiableCollection.forEach(Collections.java:1082)
	at bt.peer.PeerRegistry.collectAndVisitPeers(PeerRegistry.java:106)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.runAndReset(FutureTask.java:308)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.access$301(ScheduledThreadPoolExecutor.java:180)
	at java.util.concurrent.ScheduledThreadPoolExecutor$ScheduledFutureTask.run(ScheduledThreadPoolExecutor.java:294)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: bt.BtException: Unexpected error in peer lookup: null. Diagnostics:
==========================
DHT Diagnostics. Type IPV4_DHT
# of active servers / all servers: 1/1
-----------------------
Stats
Reachable node estimate: 2 (0.5)
DB Keys: 0
DB Items: 0
TX sum: 3 RX sum: 2
avg task time/avg 1st result time (ms): 10000/10000
Uptime: PT5.194Ss
RPC stats
### local RPCs
            Method                 REQ |                 RSP               Error             Timeout 

              PING                   3 |                   2                   0                   0 
         FIND_NODE                   0 |                   0                   0                   0 
         GET_PEERS                   0 |                   0                   0                   0 
     ANNOUNCE_PEER                   0 |                   0                   0                   0 
               GET                   0 |                   0                   0                   0 
               PUT                   0 |                   0                   0                   0 
 SAMPLE_INFOHASHES                   0 |                   0                   0                   0 
           UNKNOWN                   0 |                   0                   0                   0 

### remote RPCs
            Method                 REQ |                 RSP              Errors 

              PING                   0 |                   0                   0 
         FIND_NODE                   0 |                   0                   0 
         GET_PEERS                   0 |                   0                   0 
     ANNOUNCE_PEER                   0 |                   0                   0 
               GET                   0 |                   0                   0 
               PUT                   0 |                   0                   0 
 SAMPLE_INFOHASHES                   0 |                   0                   0 
           UNKNOWN                   0 |                   0                   0 
-----------------------
Routing table
buckets: 1 / entries: 2
all   num:2 rep:0 [Home]

-----------------------
RPC Servers
9D0499BD 5BA357B7 A140940E 8963221E 0E15CFF4	bind: /192.168.0.174 consensus: null
rx: 2 tx: 3 active: 1 baseRTT: 8025 loss: 0.490050  loss (verified): 0.500000 uptime: PT5.005S
RTT stats (2samples)  mean:2562.5002 median:125.0 mode:125.0 10tile:125.0 90tile:8025.0
     0 |     50 |    100 |    150 |    200 |    250 |    300 |    350 |    400 |    450 |    500 |    550 |    600 |    650 |    700 |    750 |    800 |    850 |    900 |    950 |   1000 |   1050 |   1100 |   1150 |   1200 |   1250 |   1300 |   1350 |   1400 |   1450 |   1500 |   1550 |   1600 |   1650 |   1700 |   1750 |   1800 |   1850 |   1900 |   1950 |   2000 |   2050 |   2100 |   2150 |   2200 |   2250 |   2300 |   2350 |   2400 |   2450 |   2500 |   2550 |   2600 |   2650 |   2700 |   2750 |   2800 |   2850 |   2900 |   2950 |   3000 |   3050 |   3100 |   3150 |   3200 |   3250 |   3300 |   3350 |   3400 |   3450 |   3500 |   3550 |   3600 |   3650 |   3700 |   3750 |   3800 |   3850 |   3900 |   3950 |   4000 |   4050 |   4100 |   4150 |   4200 |   4250 |   4300 |   4350 |   4400 |   4450 |   4500 |   4550 |   4600 |   4650 |   4700 |   4750 |   4800 |   4850 |   4900 |   4950 |   5000 |   5050 |   5100 |   5150 |   5200 |   5250 |   5300 |   5350 |   5400 |   5450 |   5500 |   5550 |   5600 |   5650 |   5700 |   5750 |   5800 |   5850 |   5900 |   5950 |   6000 |   6050 |   6100 |   6150 |   6200 |   6250 |   6300 |   6350 |   6400 |   6450 |   6500 |   6550 |   6600 |   6650 |   6700 |   6750 |   6800 |   6850 |   6900 |   6950 |   7000 |   7050 |   7100 |   7150 |   7200 |   7250 |   7300 |   7350 |   7400 |   7450 |   7500 |   7550 |   7600 |   7650 |   7700 |   7750 |   7800 |   7850 |   7900 |   7950 |   8000 |   8050 |   8100 |   8150 |   8200 |   8250 |   8300 |   8350 |   8400 |   8450 |   8500 |   8550 |   8600 |   8650 |   8700 |   8750 |   8800 |   8850 |   8900 |   8950 |   9000 |   9050 |   9100 |   9150 |   9200 |   9250 |   9300 |   9350 |   9400 |   9450 |   9500 |   9550 |   9600 |   9650 |   9700 |   9750 |   9800 |   9850 |   9900 |   9950 | 
    0% |     0% |    50% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% |     0% | 
-----------------------
Blacklist
{}
-----------------------
Lookup Cache
anchors (0):
buckets (1) / entries (0):

all entries: 0
-----------------------
Tasks
next id: 2
#### active: 
#### queued: 
NodeLookup 2 target:9D0499BD 5BA357B7 A140940E 8963221E 0E15CFF4 todo:0 srv: 9D0499BD 5BA357B7 A140940E 8963221E 0E15CFF4 QUEUED name:Bootstrap: lookup for self




	at bt.dht.MldhtService.getPeers(MldhtService.java:188)
	at bt.dht.DHTPeerSource.collectPeers(DHTPeerSource.java:49)
	at bt.peer.ScheduledPeerSource.lambda$schedulePeerCollection$0(ScheduledPeerSource.java:81)
	at java.util.concurrent.Executors$RunnableAdapter.call(Executors.java:511)
	at java.util.concurrent.FutureTask.run(FutureTask.java:266)
	... 3 more
Caused by: java.lang.NullPointerException
	at bt.dht.MldhtService.getPeers(MldhtService.java:178)
	... 7 more
[06/May/2020:12:24:01,277] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:24,995] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:27,487] ERROR DataReceivingLoop: Failed to process key
bt.protocol.InvalidMessageException: Can't create PEX message: no peers added/dropped
	at bt.peerexchange.PeerExchange.<init>(PeerExchange.java:105)
	at bt.peerexchange.PeerExchange.parse(PeerExchange.java:66)
	at bt.peerexchange.PeerExchangeMessageHandler.decode(PeerExchangeMessageHandler.java:56)
	at bt.protocol.extended.ExtendedProtocol.doDecode(ExtendedProtocol.java:136)
	at bt.protocol.handler.BaseMessageHandler.decode(BaseMessageHandler.java:99)
	at bt.protocol.StandardBittorrentProtocol.decode(StandardBittorrentProtocol.java:260)
	at bt.net.pipeline.MessageDeserializer.deserialize(MessageDeserializer.java:47)
	at bt.net.pipeline.DefaultChannelPipeline.processInboundData(DefaultChannelPipeline.java:100)
	at bt.net.pipeline.DefaultChannelPipeline.fireDataReceived(DefaultChannelPipeline.java:80)
	at bt.net.pipeline.DefaultChannelPipeline.access$000(DefaultChannelPipeline.java:30)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireDataReceived(DefaultChannelPipeline.java:217)
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:133)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:43,031] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,037] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,039] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,051] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,054] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,055] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,056] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,071] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,072] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: An established connection was aborted by the software in your host machine
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:118)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,083] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,088] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,090] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,099] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,110] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,125] WARN  MultiTracker: Unexpected error during interaction with the tracker: HttpTracker{baseUri=http://bt3.t-ru.org/ann}
java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:139)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:155)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:284)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:140)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:261)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:165)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:167)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:272)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:124)
	at org.apache.http.impl.execchain.MinimalClientExec.execute(MinimalClientExec.java:213)
	at org.apache.http.impl.client.MinimalHttpClient.doExecute(MinimalHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
	at bt.tracker.http.HttpTracker.sendEvent(HttpTracker.java:150)
	at bt.tracker.http.HttpTracker.access$000(HttpTracker.java:51)
	at bt.tracker.http.HttpTracker$1.complete(HttpTracker.java:117)
	at bt.tracker.MultiTracker$1.lambda$complete$2(MultiTracker.java:85)
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:125)
	at bt.tracker.MultiTracker$1.complete(MultiTracker.java:85)
	at bt.torrent.TrackerAnnouncer.lambda$complete$2(TrackerAnnouncer.java:98)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.torrent.TrackerAnnouncer.complete(TrackerAnnouncer.java:96)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.torrent.ProcessTorrentStage.onCompleted(ProcessTorrentStage.java:97)
	at bt.processor.torrent.ProcessTorrentStage.complete(ProcessTorrentStage.java:90)
	at bt.processor.torrent.ProcessTorrentStage.doExecute(ProcessTorrentStage.java:66)
	at bt.processor.torrent.ProcessTorrentStage.doExecute(ProcessTorrentStage.java:34)
	at bt.processor.TerminateOnErrorProcessingStage.doExecute(TerminateOnErrorProcessingStage.java:38)
	at bt.processor.RoutingProcessingStage.execute(RoutingProcessingStage.java:39)
	at bt.processor.ChainProcessor.doExecute(ChainProcessor.java:112)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:96)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.lambda$process$0(ChainProcessor.java:81)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:43,133] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.IOException: An existing connection was forcibly closed by the remote host
	at sun.nio.ch.SocketDispatcher.read0(Native Method)
	at sun.nio.ch.SocketDispatcher.read(SocketDispatcher.java:43)
	at sun.nio.ch.IOUtil.readIntoNativeBuffer(IOUtil.java:223)
	at sun.nio.ch.IOUtil.read(IOUtil.java:192)
	at sun.nio.ch.SocketChannelImpl.read(SocketChannelImpl.java:380)
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:118)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,138] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,159] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,161] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,192] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,254] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,304] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,351] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,575] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:43,659] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:24:45,384] WARN  MultiTracker: Unexpected error during interaction with the tracker: HttpTracker{baseUri=http://retracker.local/announce}
java.net.UnknownHostException: retracker.local
	at java.net.Inet6AddressImpl.lookupAllHostAddr(Native Method)
	at java.net.InetAddress$2.lookupAllHostAddr(InetAddress.java:929)
	at java.net.InetAddress.getAddressesFromNameService(InetAddress.java:1324)
	at java.net.InetAddress.getAllByName0(InetAddress.java:1277)
	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at org.apache.http.impl.conn.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:45)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:111)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
	at org.apache.http.impl.execchain.MinimalClientExec.execute(MinimalClientExec.java:183)
	at org.apache.http.impl.client.MinimalHttpClient.doExecute(MinimalHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
	at bt.tracker.http.HttpTracker.sendEvent(HttpTracker.java:150)
	at bt.tracker.http.HttpTracker.access$000(HttpTracker.java:51)
	at bt.tracker.http.HttpTracker$1.complete(HttpTracker.java:117)
	at bt.tracker.MultiTracker$1.lambda$complete$2(MultiTracker.java:85)
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:125)
	at bt.tracker.MultiTracker$1.complete(MultiTracker.java:85)
	at bt.torrent.TrackerAnnouncer.lambda$complete$2(TrackerAnnouncer.java:98)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.torrent.TrackerAnnouncer.complete(TrackerAnnouncer.java:96)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.torrent.ProcessTorrentStage.onCompleted(ProcessTorrentStage.java:97)
	at bt.processor.torrent.ProcessTorrentStage.complete(ProcessTorrentStage.java:90)
	at bt.processor.torrent.ProcessTorrentStage.doExecute(ProcessTorrentStage.java:66)
	at bt.processor.torrent.ProcessTorrentStage.doExecute(ProcessTorrentStage.java:34)
	at bt.processor.TerminateOnErrorProcessingStage.doExecute(TerminateOnErrorProcessingStage.java:38)
	at bt.processor.RoutingProcessingStage.execute(RoutingProcessingStage.java:39)
	at bt.processor.ChainProcessor.doExecute(ChainProcessor.java:112)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:96)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.lambda$process$0(ChainProcessor.java:81)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:45,384] ERROR TrackerAnnouncer: Failed to announce 'complete' event due to unexpected error during interaction with the tracker: MultiTracker{trackerTiers=[[HttpTracker{baseUri=http://bt3.t-ru.org/ann}], [HttpTracker{baseUri=http://retracker.local/announce}]]}
bt.BtException: All trackers failed; responses (in chrono order): [bt.tracker.TrackerResponse@3255e5c1, bt.tracker.TrackerResponse@23ee288a]
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:145)
	at bt.tracker.MultiTracker$1.complete(MultiTracker.java:85)
	at bt.torrent.TrackerAnnouncer.lambda$complete$2(TrackerAnnouncer.java:98)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.torrent.TrackerAnnouncer.complete(TrackerAnnouncer.java:96)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.torrent.ProcessTorrentStage.onCompleted(ProcessTorrentStage.java:97)
	at bt.processor.torrent.ProcessTorrentStage.complete(ProcessTorrentStage.java:90)
	at bt.processor.torrent.ProcessTorrentStage.doExecute(ProcessTorrentStage.java:66)
	at bt.processor.torrent.ProcessTorrentStage.doExecute(ProcessTorrentStage.java:34)
	at bt.processor.TerminateOnErrorProcessingStage.doExecute(TerminateOnErrorProcessingStage.java:38)
	at bt.processor.RoutingProcessingStage.execute(RoutingProcessingStage.java:39)
	at bt.processor.ChainProcessor.doExecute(ChainProcessor.java:112)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:96)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.lambda$process$0(ChainProcessor.java:81)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:45,439] WARN  MultiTracker: Unexpected error during interaction with the tracker: HttpTracker{baseUri=http://bt3.t-ru.org/ann}
java.net.SocketException: Connection reset
	at java.net.SocketInputStream.read(SocketInputStream.java:210)
	at java.net.SocketInputStream.read(SocketInputStream.java:141)
	at org.apache.http.impl.io.SessionInputBufferImpl.streamRead(SessionInputBufferImpl.java:139)
	at org.apache.http.impl.io.SessionInputBufferImpl.fillBuffer(SessionInputBufferImpl.java:155)
	at org.apache.http.impl.io.SessionInputBufferImpl.readLine(SessionInputBufferImpl.java:284)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:140)
	at org.apache.http.impl.conn.DefaultHttpResponseParser.parseHead(DefaultHttpResponseParser.java:57)
	at org.apache.http.impl.io.AbstractMessageParser.parse(AbstractMessageParser.java:261)
	at org.apache.http.impl.DefaultBHttpClientConnection.receiveResponseHeader(DefaultBHttpClientConnection.java:165)
	at org.apache.http.impl.conn.CPoolProxy.receiveResponseHeader(CPoolProxy.java:167)
	at org.apache.http.protocol.HttpRequestExecutor.doReceiveResponse(HttpRequestExecutor.java:272)
	at org.apache.http.protocol.HttpRequestExecutor.execute(HttpRequestExecutor.java:124)
	at org.apache.http.impl.execchain.MinimalClientExec.execute(MinimalClientExec.java:213)
	at org.apache.http.impl.client.MinimalHttpClient.doExecute(MinimalHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
	at bt.tracker.http.HttpTracker.sendEvent(HttpTracker.java:150)
	at bt.tracker.http.HttpTracker.access$000(HttpTracker.java:51)
	at bt.tracker.http.HttpTracker$1.stop(HttpTracker.java:112)
	at bt.tracker.MultiTracker$1.lambda$stop$1(MultiTracker.java:80)
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:125)
	at bt.tracker.MultiTracker$1.stop(MultiTracker.java:80)
	at bt.torrent.TrackerAnnouncer.lambda$stop$1(TrackerAnnouncer.java:88)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.torrent.TrackerAnnouncer.stop(TrackerAnnouncer.java:86)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.torrent.TorrentContextFinalizer.finalizeContext(TorrentContextFinalizer.java:37)
	at bt.processor.torrent.TorrentContextFinalizer.finalizeContext(TorrentContextFinalizer.java:24)
	at bt.processor.ChainProcessor.lambda$doExecute$2(ChainProcessor.java:134)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.ChainProcessor.doExecute(ChainProcessor.java:134)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:96)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.lambda$process$0(ChainProcessor.java:81)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:45,440] WARN  MultiTracker: Unexpected error during interaction with the tracker: HttpTracker{baseUri=http://retracker.local/announce}
java.net.UnknownHostException: retracker.local
	at java.net.InetAddress.getAllByName0(InetAddress.java:1281)
	at java.net.InetAddress.getAllByName(InetAddress.java:1193)
	at java.net.InetAddress.getAllByName(InetAddress.java:1127)
	at org.apache.http.impl.conn.SystemDefaultDnsResolver.resolve(SystemDefaultDnsResolver.java:45)
	at org.apache.http.impl.conn.DefaultHttpClientConnectionOperator.connect(DefaultHttpClientConnectionOperator.java:111)
	at org.apache.http.impl.conn.PoolingHttpClientConnectionManager.connect(PoolingHttpClientConnectionManager.java:353)
	at org.apache.http.impl.execchain.MinimalClientExec.execute(MinimalClientExec.java:183)
	at org.apache.http.impl.client.MinimalHttpClient.doExecute(MinimalHttpClient.java:107)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:71)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:220)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:164)
	at org.apache.http.impl.client.CloseableHttpClient.execute(CloseableHttpClient.java:139)
	at bt.tracker.http.HttpTracker.sendEvent(HttpTracker.java:150)
	at bt.tracker.http.HttpTracker.access$000(HttpTracker.java:51)
	at bt.tracker.http.HttpTracker$1.stop(HttpTracker.java:112)
	at bt.tracker.MultiTracker$1.lambda$stop$1(MultiTracker.java:80)
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:125)
	at bt.tracker.MultiTracker$1.stop(MultiTracker.java:80)
	at bt.torrent.TrackerAnnouncer.lambda$stop$1(TrackerAnnouncer.java:88)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.torrent.TrackerAnnouncer.stop(TrackerAnnouncer.java:86)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.torrent.TorrentContextFinalizer.finalizeContext(TorrentContextFinalizer.java:37)
	at bt.processor.torrent.TorrentContextFinalizer.finalizeContext(TorrentContextFinalizer.java:24)
	at bt.processor.ChainProcessor.lambda$doExecute$2(ChainProcessor.java:134)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.ChainProcessor.doExecute(ChainProcessor.java:134)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:96)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.lambda$process$0(ChainProcessor.java:81)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:45,441] ERROR TrackerAnnouncer: Failed to announce 'stop' event due to unexpected error during interaction with the tracker: MultiTracker{trackerTiers=[[HttpTracker{baseUri=http://bt3.t-ru.org/ann}], [HttpTracker{baseUri=http://retracker.local/announce}]]}
bt.BtException: All trackers failed; responses (in chrono order): [bt.tracker.TrackerResponse@a8913de, bt.tracker.TrackerResponse@11ad3b3b]
	at bt.tracker.MultiTracker$1.tryForAllTrackers(MultiTracker.java:145)
	at bt.tracker.MultiTracker$1.stop(MultiTracker.java:80)
	at bt.torrent.TrackerAnnouncer.lambda$stop$1(TrackerAnnouncer.java:88)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.torrent.TrackerAnnouncer.stop(TrackerAnnouncer.java:86)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.torrent.TorrentContextFinalizer.finalizeContext(TorrentContextFinalizer.java:37)
	at bt.processor.torrent.TorrentContextFinalizer.finalizeContext(TorrentContextFinalizer.java:24)
	at bt.processor.ChainProcessor.lambda$doExecute$2(ChainProcessor.java:134)
	at java.util.Optional.ifPresent(Optional.java:159)
	at bt.processor.ChainProcessor.doExecute(ChainProcessor.java:134)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:96)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.executeStage(ChainProcessor.java:98)
	at bt.processor.ChainProcessor.lambda$process$0(ChainProcessor.java:81)
	at java.util.concurrent.CompletableFuture$AsyncRun.run(CompletableFuture.java:1640)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
[06/May/2020:12:24:58,309] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
[06/May/2020:12:26:02,333] ERROR DataReceivingLoop: Failed to process key
java.lang.RuntimeException: Unexpected I/O error
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:83)
	at bt.net.pipeline.DefaultChannelPipeline$DefaultChannelHandlerContext.fireChannelReady(DefaultChannelPipeline.java:192)
	at bt.net.DataReceivingLoop.processKey(DataReceivingLoop.java:161)
	at bt.net.DataReceivingLoop.run(DataReceivingLoop.java:126)
	at java.util.concurrent.ThreadPoolExecutor.runWorker(ThreadPoolExecutor.java:1149)
	at java.util.concurrent.ThreadPoolExecutor$Worker.run(ThreadPoolExecutor.java:624)
	at java.lang.Thread.run(Thread.java:748)
Caused by: java.io.EOFException
	at bt.net.pipeline.SocketChannelHandler.processInboundData(SocketChannelHandler.java:136)
	at bt.net.pipeline.SocketChannelHandler.read(SocketChannelHandler.java:80)
	... 6 more
